druid-shell/src/platform/x11/window.rs

```diff
@@ -1669,3 +1694,59 @@ unsafe impl HasRawWindowHandle for WindowHandle {
         RawWindowHandle::Xcb(handle)
     }
 }
+
+fn make_cursor(
+    conn: &XCBConnection,
+    root_window: u32,
+    argb32_format: Pictformat,
+    desc: &CursorDesc,
+) -> Result<Cursor, ReplyOrIdError> {
+    // BEGIN: Lots of code just to get the image into a RENDER Picture
+
+    // No idea how to sanely get the pixel values, so I'll go with 'insane':
```

Comment:
[`ImageBuf::raw_pixels`](https://docs.rs/piet/0.4.0/piet/struct.ImageBuf.html#method.raw_pixels) should work

Reply:
Yeah, but then I need to handle all the different possible `ImageFormat` values. The GTK backend does not do that (I guess it only handles two of the four possible values properly), so this seems complicated/untested.

And with `raw_pixels`, `RgbaSeparate` still needs to be converted to premultiplied "by hand". The simplest way for that could be the already existing code, I guess.